test/integration/framework/utils.go

```diff
@@ -0,0 +1,358 @@
+
+	var podList *corev1.PodList
+	var err error
+
+	switch seedOrShoot {
+	case "seed":
+		podList, err = o.K8sSeedClient.ListPods(o.Shoot.SeedNamespace, metav1.ListOptions{
+			LabelSelector: set.AsSelector().String(),
+		})
+		if err != nil {
+			return nil, err
+		}
+	case "shoot":
+		podList, err = o.K8sShootClient.ListPods(metav1.NamespaceSystem, metav1.ListOptions{
+			LabelSelector: set.AsSelector().String(),
+		})
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	// Fetch the first running pod of the Kube-apiserver
+	if len(podList.Items) > 0 {
+		for _, pod := range podList.Items {
+			o.ShootGardenerTest.Logger.Infof("%s:%s", pod.Namespace, pod.Name)
+			if pod.Status.Phase == corev1.PodRunning {
+				runningPodWithDesiredLabels = &pod
+				break
+			}
+		}
+		return runningPodWithDesiredLabels, nil
+	}
+
+	return nil, nil
+}
+
+// getPodByLabels fetches the first running pod with the desired set of labels <labelsMap>
+func (o *GardenerTestOperation) getPodByLabels(labelsMap map[string]string, namespace string, seedOrShoot string) (*corev1.Pod, error) {
+	var runningPodWithDesiredLabels *corev1.Pod
+	set := labels.Set(labelsMap)
+
+	var podList *corev1.PodList
+	var err error
+
+	switch seedOrShoot {
+	case "seed":
+		podList, err = o.K8sSeedClient.ListPods(o.Shoot.SeedNamespace, metav1.ListOptions{
+			LabelSelector: set.AsSelector().String(),
+		})
+		if err != nil {
+			return nil, err
+		}
+	case "shoot":
+		podList, err = o.K8sShootClient.ListPods(namespace, metav1.ListOptions{
+			LabelSelector: set.AsSelector().String(),
+		})
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	// Fetch the first running pod of the Kube-apiserver
+	if len(podList.Items) > 0 {
+		pod := podList.Items[0]
+		o.ShootGardenerTest.Logger.Infof("%s:%s", pod.Namespace, pod.Name)
+		runningPodWithDesiredLabels = &pod
+	}
+	return runningPodWithDesiredLabels, nil
+
+}
+
+// GetAdminPassword gets the admin password for authenticating agains the api
+func (o *GardenerTestOperation) getAdminPassword() (*string, error) {
+	password, err := getPasswordFromSecret(o.K8sSeedClient, o.Shoot.SeedNamespace, "kubecfg")
+	if err != nil {
+		return nil, err
+	}
+	return &password, nil
+}
+
+func (s *ShootGardenerTest) mergePatch(oldShoot, newShoot []byte) error {
+	patchBytes, err := strategicpatch.CreateTwoWayMergePatch(oldShoot, newShoot, v1beta1.Shoot{})
+	if err != nil {
+		return fmt.Errorf("failed to patch bytes")
+	}
+
+	_, err = s.K8sGardenClient.Garden().GardenV1beta1().Shoots(s.Shoot.Namespace).Patch(s.Shoot.Name, types.StrategicMergePatchType, patchBytes)
+	if err != nil {
+		return fmt.Errorf("failed to update shoot with new value: %s", err.Error())
+	}
+
+	return nil
+}
+
+func shootCreationCompleted(newStatus *v1beta1.ShootStatus) bool {
+	validConditionMatrix := true
+	if len(newStatus.Conditions) > 0 {
+		for _, condition := range newStatus.Conditions {
+			validConditionMatrix = validConditionMatrix && (condition.Status == "True")
```

Comment:
For `"True"` there is a constant.

Reply:
Where is that defined?

Reply:
Still unresolved.

Reply:
it was resolved, in the `deploymentCreated` method but not in the `shootCreated` method. Got confused  for a sec and forgot it's in two places.